[Remove SitesList] Move `lib/site/computed-attributes` to appropriate state folder

Tug opened the pull request:
This PR removes the reference of `lib/site/computed-attributes` from `sites/selectors` by adding the `getSiteOptions` and the `isSiteCustomizable` selectors. The original `getComputedAttributes` applied to `Site` instances is not removed yet since we still have a lot of code using SitesList and accessing those computed properties through those objects.

This PR has some additional changes:
- It updates any code calling the `Site#isCustomizable()` method and replaces it with the `isSiteCustomizable` selector.
- The bug fixed in #11301 was still calling `Site#getUser()`, I switched to using the `UsersStore` and added a new method to wpcom.js undocumented to fetch a single user from its `userId`.
- `Site#fetchUsers`, `Site#getUser()` and `Site#getUser()` are removed since they aren't used anywhere now in calypso.

## Testing Instructions
- `git checkout update/computed-attributes && make run`
- Navigate to http://calypso.localhost:3000

### Testing the bug fix
- Select a site, go to `Sharing > Connections` and open a connection set by another user than yourself
- assert that you can see `CONNECTED BY <login name of the user who set the connection>` next to the username connected

### Testing `isSiteCustomizable`
- Run the tests (`make test`)
- Check that you don't have access to `Themes` in the left sidebar when you don't have the permissions

### Testing the removal of getUsers methods in Site
- Go to `People` and check that you can see the users as usual
- Check that the app works for different kind of sites (jetpack, wpcom with mapped domain...)


## Reviews
- [ ] Product review
- [ ] Code review

retrofox commented on client/lib/users/actions.js:
```diff
@@ -157,8 +157,34 @@ const UsersActions = {
+					error
+				} );
+			} else {
+				Dispatcher.handleServerAction( {
```
Maybe add remove `else` and add a `return`?

```js
if ( error ) {
  return Dispatcher. handleServerAction( {
    // ...
  } )
}

Dispatcher.handleServerAction( {
  // ...
} );
```

retrofox commented on client/lib/wpcom-undocumented/lib/site.js:
```diff
@@ -151,7 +151,11 @@ UndocumentedSite.prototype.updateUser = function( userId, attributes, callback )
 
-UndocumentedSite.prototype.getUser = function( login, callback ) {
+UndocumentedSite.prototype.getUser = function( userId, callback ) {
+	return this.wpcom.req.get( '/sites/' + this._id + '/users/' + userId, callback );
```
maybe add string templates here?

```js
return this.wpcom.req.get( `/sites/${ this._id }/users/${ userId }`, callback );
```

Feel free to ignore this comment. It isn't a blocker.

retrofox commented on client/my-sites/people/controller.js:
```diff
@@ -126,7 +126,7 @@ function renderSingleTeamMember( context ) {
 
 		if ( ! user ) {
-			UsersActions.fetchUser( { siteId: siteId }, userLogin );
+			UsersActions.fetchUserByLogin( { siteId: siteId }, userLogin );
```
You could remove the object assignment here since the `key` has the same name that the var which has the value of the key (`siteId`).

```js
UsersActions.fetchUserByLogin( { siteId }, userLogin );
```

Again, it isn't a blocker.

retrofox commented on client/my-sites/sharing/connections/connection.jsx:
```diff
@@ -203,7 +228,7 @@ export default connect(
 			siteId,
 			userHasCaps: canCurrentUser( state, siteId, 'edit_others_posts' ),
-			userId: getCurrentUserId( state ),
+			userId: getCurrentUserId( state )
```
AFAIK it isn't needed. In fact I'm not sure if we should keep or not the comma here.

dmsnell replied:
although we don't have this in our style guide, I highly encourage adding the trailing comma and I think a few others do as well. it certainly makes for less confusing diffs

Tug commented:
Thanks @retrofox and @dmsnell for having a look at it. I addressed all your comments except the one about `wpcom-undocumented/lib/site.js` which would mean mixing both syntax in the way we build urls in this lib. I don't mind much but I think this file globally looks better without this change.

yurynix commented on client/my-sites/sharing/connections/connection.jsx:
```diff
@@ -78,12 +79,35 @@ class SharingConnection extends Component {
+	}
+
+	componentWillReceiveProps( nextProps, prevProps ) {
+		if ( nextProps.siteId !== prevProps.siteId || nextProps.connection !== prevProps.connection ) {
```
Maybe I don't understand that right, and sorry if I am, but while in [componentWillReceiveProps](https://facebook.github.io/react/docs/react-component.html#componentwillreceiveprops) `prevProps` are `this.props`, so `fetchKeyringUser` is actually fetching the data for `prevProps` not `nextProps` as it should be?

dmsnell replied:
cool catch @yurynix! `prevProps` here should be `undefined` according to the React docs…

Tug replied:
Indeed! thanks for noticing @yurynix !

gwwar commented:
Any blockers on this one @retrofox @yurynix ? Or does this just need another look through and testing?

dmsnell commented on client/lib/people/log-store.js:
```diff
@@ -110,7 +110,11 @@ PeopleLogStore.dispatchToken = Dispatcher.register( function( payload ) {
+				action.type,
+				action.fetchOptions.siteId,
+				{ login: action.login, userId: action.userId },
+				action.error.error );
```
style: this parenthesis should be on the next line indented with `addLog(`

yurynix commented:
@Tug @gwwar 
I think that one needs addressing: https://github.com/Automattic/wp-calypso/pull/11310#discussion_r102656888

Tug commented:
Rebased and addressed your comments, I believe this is ready for a final check :)

dmsnell commented:
@Tug can we get a status update on this PR? Is it still relevant?

Tug commented:
@dmsnell It looks like it's still relevant. I just rebased and moved `isSiteCustomizable()` to `state/selectors`.

There are a few different changes in the same PR but I think it's still manageable. Would be great if someone would give it a final review!

dmsnell commented on client/lib/users/actions.js:
```diff
@@ -113,52 +113,78 @@ const UsersActions = {
+				action: 'UPDATE_SITE_USER',
+				siteId: siteId,
+				user: user,
+				data: data
```
fun fact! we can use object shorthand notation here to prevent repeating ourselves

```js
{
	type: '…',
	action: '…',
	siteId,
	user,
	data,
}
```

Tug commented:
@dmsnell I only picked the changes about the selectors from this PR and made a new one, it should be a lot clearer now https://github.com/Automattic/wp-calypso/pull/13160.
Will create 2 more PRs for the fixes and the cleanup later.
